documentation-site/components/yard/yard.tsx

```diff
@@ -175,10 +138,50 @@ export default withRouter(
           themeName: theme.name,
         }),
       codeNoRecompile: '',
-      props: initialUrlProps || propsConfig,
-      theme: initialUrlTheme || componentThemeObj,
+      props: propsConfig,
+      theme: componentThemeObj,
     });
 
+    React.useEffect(() => {
+      // initialize from the URL
+      if (router.query.code && !urlCodeHydrated) {
+        setUrlCodeHydrated(true);
+        try {
+          const propValues: {[key: string]: any} = {};
+          const {parsedProps, parsedTheme} = parseCode(
+            router.query.code as string,
+            componentName,
+          );
+          Object.keys(propsConfig).forEach(name => {
+            //@ts-ignore
+            propValues[name] = propsConfig[name].value;
+            if (name === 'overrides') {
+              // overrides need a special treatment since the value needs to
+              // be further analyzed and parsed
+              propValues[name] = parseOverrides(
+                parsedProps[name],
+                propsConfig.overrides && propsConfig.overrides.meta
+                  ? propsConfig.overrides.meta.names || []
+                  : [],
+              );
+            } else {
+              propValues[name] = parsedProps[name];
+            }
+          });
+          dispatch({
+            type: Action.Update,
+            payload: {
+              code: router.query.code,
+              updatedPropValues: propValues,
+              theme: parsedTheme,
+            },
+          });
+        } catch (e) {
+          console.log(e);
```

Comment:
should we use `console.error`?

Reply:
yea, I'll probably remove those completely